src/App.tsx

```diff
@@ -15,48 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType: SortType, isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const toShowResetButton = isReversed || sortType !== SortType.NONE;
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
```

Comment:
I'd rather put a space in between, like 2 groups – hooks and other variables
```suggestion
  const [isReversed, setIsReversed] = useState(false);
  const [sortType, setSortType] = useState(SortType.NONE);

  const toShowResetButton = isReversed || sortType !== SortType.NONE;
  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
```